### PR TITLE
Add the openhim channel for the covid19 case outcome

### PR DESCRIPTION
**Pull request by bradsawadye**
This is the channel that handles the storing of the Covid19 case outcome in FHIR and DHIS2

DSC19-145 DSC19-111

### CODE REVIEW

File: docker/importer/volume/surveillance-mediator/case-outcome/endpoint-flattener.json [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    "transforms.itemsObject.admission_date.answer.valueDateTime": "encounters[1].period.start",
+    "constants.admission_date_encounter_status": "encounters[1].status",
+
+    "transforms.observationId0":"observations[0].id",
+    "transforms.itemsObject.outcome_asymp.answer[].valueCoding": "observations[0].value.valueCodeableConcept.coding[]",
+    "constants.outcomeAsympCoding": "observations[0].coding",
+
+    "transforms.observationId1":"observations[1].id",
+    "transforms.itemsObject.outcome_asymp_date.answer[].valueDateTime": "observations[1].value.effectiveDateTime",
+    "constants.outcomeAsympDateCoding": "observations[1].coding",
```
Comment (MattyJ007) [inline docker/importer/volume/surveillance-mediator/case-outcome/endpoint-flattener.json]: The effectiveDateTime field should be part of the Observation with the valueCoding - it's not a separate chunk of info.

File: docker/importer/volume/surveillance-mediator/case-outcome/endpoint-flattener.json [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    "transforms.itemsObject.outcome_patcourse_status_other.answer[].valueString": "observations[7].value.valueString",
+    "constants.patcourseStatusOtherCoding": "observations[7].coding",
+
+    "transforms.observationId8":"observations[8].id",
+    "transforms.itemsObject.outcome_lab_date.answer[].valueDateTime": "observations[8].value.effectiveDateTime",
+    "constants.outcomeLabDateCoding": "observations[8].coding",
+
+    "transforms.observationId9":"observations[9].id",
+    "transforms.itemsObject.outcome_lab_result.answer[].valueCoding": "observations[9].value.valueCodeableConcept.coding[]",
+    "constants.outcomeLabResultCoding": "observations[9].coding",
```
Comment (MattyJ007) [inline docker/importer/volume/surveillance-mediator/case-outcome/endpoint-flattener.json]: This data relates to one Observation

**Review by MattyJ007 — CHANGES_REQUESTED** [review]
Looking great - just two Observations that need to be merged together.
Very good commit message 💯

File: docker/importer/volume/surveillance-mediator/case-outcome/endpoint-flattener.json (hunk 2 above)
Comment (bradsawadye) [reply docker/importer/volume/surveillance-mediator/case-outcome/endpoint-flattener.json]: Addressed

File: docker/importer/volume/surveillance-mediator/case-outcome/endpoint-flattener.json (hunk 1 above)
Comment (bradsawadye) [reply docker/importer/volume/surveillance-mediator/case-outcome/endpoint-flattener.json]: Addressed

File: docker/importer/volume/surveillance-mediator/case-outcome/endpoint-composition.json [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    "compositionId": "'urn:uuid:' & $substringAfter($string($random()), '0.')",
+    "hospitalEncounter" : "requestBody.data.encounters[0].id",
+    "caseEncounter": "requestBody.data.encounters[0].id"
+  },
+  "inputMapping": {
+    "constants.template": "composition",
+    "transforms.compositionId": "composition.fullUrl",
+    "requestBody.author": "composition.resource.author[]+",
+    "requestBody.data.identifier": "composition.resource.identifier.value",
+    "requestBody.compositionDate": "composition.resource.date",
```
Comment (MattyJ007) [inline docker/importer/volume/surveillance-mediator/case-outcome/endpoint-composition.json]: @bradsawadye This date field doesn't appear to be coming through in the final Composition
```suggestion
    "requestBody.data.compositionDate": "composition.resource.date",
```

I think it's the nesting

**Review by MattyJ007 — APPROVED** [review]
Looks good

**Pull request merged**